vnext/Shared/MemoryMappedBuffer.cpp

```diff
@@ -0,0 +1,154 @@
+          FILE_SHARE_READ,
+          OPEN_EXISTING,
+          nullptr /* pCreateExParams */),
+      &CloseHandle};
+#else
+  std::unique_ptr<void, decltype(&CloseHandle)> fileHandle{
+      CreateFileW(
+          filename,
+          GENERIC_READ,
+          FILE_SHARE_READ,
+          nullptr /* lpSecurityAttributes */,
+          OPEN_EXISTING,
+          FILE_ATTRIBUTE_NORMAL,
+          nullptr /* hTemplateFile */),
+      &CloseHandle};
+#endif
+
+  if (fileHandle.get() == INVALID_HANDLE_VALUE) {
+    throw facebook::jsi::JSINativeException(
+        "CreateFileW/CreateFile2 failed with last error " +
+        std::to_string(GetLastError()));
+  }
+
+  LARGE_INTEGER size;
+  if (!GetFileSizeEx(fileHandle.get(), &size)) {
+    throw facebook::jsi::JSINativeException(
+        "GetFileSizeEx failed with last error " +
+        std::to_string(GetLastError()));
+  }
+
+  if (size.LowPart == 0) {
+    throw facebook::jsi::JSINativeException("Cannot memory map an empty file.");
+  }
+
+  if (size.HighPart != 0) {
+    throw facebook::jsi::JSINativeException(
+        "MemoryMappedBuffer only support files whose size can fit within an "
+        "uint32_t.");
+  }
+
+  m_size = size.LowPart;
+  if (offset > m_size) {
+    m_size = 0;
+    throw facebook::jsi::JSINativeException("Invalid offset.");
+  }
+  m_offset = offset;
+
+#if (defined(WINRT))
+  m_fileMapping.reset(CreateFileMappingFromApp(
+      fileHandle.get(),
+      nullptr /* SecurityAttributes */,
+      PAGE_READONLY,
+      m_size,
+      nullptr /* Name */));
+#else
+  m_fileMapping.reset(CreateFileMapping(
+      fileHandle.get(),
+      nullptr /* lpAttributes */,
+      PAGE_READONLY,
+      0 /* dwMaximumSizeHigh */,
+      m_size,
+      nullptr /* lpName */));
+#endif
+
+  if (!m_fileMapping) {
+    m_size = m_offset = 0;
+    throw facebook::jsi::JSINativeException(
+        "CreateFileMapping/CreateFileMappingFromApp failed with last error " +
+        std::to_string(GetLastError()));
+  }
+
+#if (defined(WINRT))
+  m_fileData.reset(MapViewOfFileFromApp(
+      m_fileMapping.get(),
+      FILE_MAP_READ,
+      0 /* FileOffset */,
+      0 /* NumberOfBytesToMap */));
+#else
+  m_fileData.reset(MapViewOfFile(
+      m_fileMapping.get(),
+      FILE_MAP_READ,
+      0 /* dwFileOffsetHigh */,
+      0 /* dwFileOffsetLow */,
+      0 /* dwNumberOfBytesToMap */));
+#endif
+
+  if (!m_fileData) {
+    m_size = m_offset = 0;
+    throw facebook::jsi::JSINativeException(
+        "MapViewOfFile/MapViewOfFileFromApp failed with last error " +
+        std::to_string(GetLastError()));
+  }
+
+  WerRegisterMemoryBlock(m_fileData.get(), m_size);
+}
+
+size_t MemoryMappedBuffer::size() const {
+  return m_size - m_offset;
+}
+
+const uint8_t *MemoryMappedBuffer::data() const {
+  return static_cast<const uint8_t *>(m_fileData.get()) + m_offset;
+}
+
+} // anonymous namespace
+
+namespace Microsoft::JSI {
+
+std::shared_ptr<facebook::jsi::Buffer> makeMemoryMappedBuffer(
```

Comment:
>makeMemoryMappedBuffer [](start = 39, length = 22)

(nit) MakeMemoryMappedBuffer

Reply:
Facebook's convention is to use camelCase instead of PascalCase for function names, and that is the convention I followed.

---
In reply to: [320884862](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r320884862) [](ancestors = 320884862)

Reply:
In my experience, we've been using PascalCase for function names.
As far as I understand, we don't need to follow Facebook's conventions for the new code we are developing, but only when modifying their code bases (i.e. `react-native`).